xonsh/commands_cache.py

```diff
@@ -232,15 +232,46 @@ def predict_threadable(self, cmd):
                 if pre in predictors:
                     predictors[name] = predictors[pre]
         if name not in predictors:
-            predictors[name] = self.default_predictor(name, cmd[0])
+            predictors[name] = self.default_predictor(name, cmd0)
         predictor = predictors[name]
+        return predictor
+
+    def predict_threadable(self, cmd):
+        """Predicts whether a command list is able to be run on a background
+        thread, rather than the main thread.
+        """
+        predictor = self.predictor_threadable(cmd[0])
         return predictor(cmd[1:])
 
     #
     # Background Predictors (as methods)
     #
 
     def default_predictor(self, name, cmd0):
+        """Default predictor, using predictor from original command if the
+        command is an alias, elseif build a predictor based on binary analysis
+        on POSIX, else return predict_true.
+        """
+        # alias stuff
+        if not os.path.isabs(cmd0) and os.sep not in cmd0:
+            alss = getattr(builtins, "aliases", dict())
+            if cmd0 in alss:
+                alias_recursion_limit = 10
+                first_args = []
+                while cmd0 in alss:
```

Comment:
I think a recursive alias resolver would be a generally useful thing -- what about breaking that out into a separate function and then using it here?